### PR TITLE
Generate stats per build

### PR DESCRIPTION
**Pull request by bishwei**
Separated these changes out from #146 

This PR adds a step to our build so that we generate a `stats.json` file that contains file sizes for all generated bundles (gestalt.css, gestalt.js, and gestalt.es.js).

Also updated our buildkite pipeline to introduce a build step that runs `yarn build` and uploads the stats file as an artifact

`stats.json`
```
{
  gestalt.js: {
    fileName: "gestalt.js",
    size: 264608,
    gzipSize: 60082
  },
  gestalt.css: {
    fileName: "gestalt.css",
    size: 53893,
    gzipSize: 9207
  },
  gestalt.es.js: {
    fileName: "gestalt.es.js",
    size: 262098,
    gzipSize: 59667
  }
}
```

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 28ba17db07d9e2a8d2e0d99df4c04c8702e74047

https://deploy-preview-160--gestalt.netlify.com

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit c004c398df2e18d70ed68350486044b18d835ed9

https://deploy-preview-160--gestalt.netlify.com

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 4b34dcd8b9f04083d1da5f156cd5b039f7bcfad0

https://deploy-preview-160--gestalt.netlify.com

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit c1d6eca9da0419ff5a269b509f19921650f8750c

https://deploy-preview-160--gestalt.netlify.com

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -60,6 +60,16 @@ steps:
...
+  - name: ":hammer_and_wrench: build"
+    command: yarn build
+    artifact_paths: 'dist/stats.json'
+    timeout_in_minutes: 10
+    agents:
+      queue: elastic
+    plugins:
+      docker-compose:
+        run: gestalt
+  - wait
```
Comment (bishwei) [inline .buildkite/pipeline.yml]: added this wait here because the `danger` step is going to rely on the stats.json file being uploaded

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**